stan/math/rev/core/vari.hpp

```diff
@@ -21,27 +43,26 @@ class var;
  * value. It also stores the adjoint for storing the partial
  * derivative with respect to the root of the derivative tree.
  *
- * The chain() method applies the chain rule. Concrete extensions
- * of this class will represent base variables or the result
- * of operations such as addition or subtraction. These extended
- * classes will store operand variables and propagate derivative
- * information via an implementation of chain().
  */
-class vari {
+template <typename T>
+class vari_value<T, std::enable_if_t<std::is_floating_point<T>::value>>
+    : public vari_base {
  private:
-  friend class var;
+  template <typename, typename>
+  friend class var_value;
 
  public:
+  using Scalar = T;
```

Comment:
Well last time I commented here it became a pretty long tangent lol, but we're sure there's a reason for both a `Scalar` and a `value_type` member? Are these to handle `scalar_type` and `value_type` implementations?

Reply:
We had `Scalar` because that's what Eigen uses. I put both because we tend to work with `std::vector`s and eigen types which call the inner type two different things and I'm fine with having both